app/Mails/Template/Templates.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Mails\Template;
+
+use Illuminate\Support\Collection;
+use InvalidArgumentException;
+
+/**
+ * @extends Collection<string, Template>
+ */
+final class Templates extends Collection
+{
+    public static function build(array $mails): self
```

Comment:
Nit: maybe use `$templates` as the name here? I see a mix of 3 names now, the class is called `Templates`, the input parameter is called `$mails` and a single item is called `$config`